Assets/Scripts/DemonMovement.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DemonMovement : MonoBehaviour
+{
+    public Transform startPoint;
+    public Transform endPoint;
+    public float speed = 5.0f;
+    private int destination  = 0;
+    // Update is called once per frame
+    void Update()
+    {
+        if (destination == 0)
+        {
+            transform.position = Vector2.MoveTowards(transform.position, startPoint.position, speed * Time.deltaTime);
```

Comment:
Use the floor prefab and apply the TornadoMovement script to Demon. TornadoMovement can be renamed to EnemyMovement and used for both tornado and demons. When enemy collides with edge in floor prefab direction gets reversed. This script can be deleted

Reply:
Implemented